hub/api/tests/test_api.py

```diff
@@ -60,12 +66,29 @@ def test_populate_dataset(ds):
 
 
 @parametrize_all_dataset_storages
-def test_compute_tensor(ds):
+def test_compute_fixed_tensor(ds):
     ds.create_tensor("image")
     ds.image.extend(np.ones((32, 28, 28)))
     np.testing.assert_array_equal(ds.image.numpy(), np.ones((32, 28, 28)))
 
 
+@parametrize_all_dataset_storages
+def test_compute_dynamic_tensor(ds):
+    ds.create_tensor("image")
+
+    a1 = np.ones((32, 28, 28))
+    a2 = np.ones((10, 36, 11))
+
+    ds.image.extend(a1)
+    ds.image.extend(a2)
+
+    expected_list = [*a1, *a2]
+    actual_list = ds.image.numpy(aslist=True)
+
+    for a1, a2 in zip(expected_list, actual_list):
+        assert (a1 == a2).all()
```

Comment:
[np.testing.assert_array_equal](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html) is preferred here. It does some extra work to check the shape, gives a better assertion message on failure, and handles the fact that `np.nan != np.nan`.